src/onnx/onnx.cpp

```diff
@@ -697,6 +795,15 @@ struct onnx_parser
             std::vector<instruction_ref> args;
             for(auto&& input : node.input())
             {
+                // For RNN, LSTM, and GRU operators, one of the input arguments
+                // is prim::Undefined, and it is ignored by protobuf. We use a
+                // hack to ignore this argument for these three operators
```

Comment:
There is no way to check for prim::Undefined?

Reply:
Looks like the model from model.ParseFromIstream(&is) ignores the prim::Undefied(). I printed out all the nodes from model, there is no such node. Then the input in the above for loop is empty.